frontend/src/services/fetchData.ts

```diff
@@ -9,33 +9,43 @@ import {
 } from '@/services/ApiServiceFactory';
 import FamLoginUserState from '@/store/FamLoginUserState';
 import { selectedApplicationId } from '@/store/ApplicationState';
+import { isNewAccess } from './utils';
 
 // --- Fetching data (from backend)
 
 export const fetchUserRoleAssignments = async (
-    applicationId: number | undefined
+    applicationId: number | undefined,
+    newUserAccessIds: string = ''
 ): Promise<FamApplicationUserRoleAssignmentGet[]> => {
     if (!applicationId) return [];
 
+    const newUsersAccessIdsList = newUserAccessIds.split(',');
+
     const userRoleAssignments = (
         await AppActlApiService.applicationsApi.getFamApplicationUserRoleAssignment(
             applicationId
         )
     ).data;
 
-    // Default sorting
-    userRoleAssignments.sort((first, second) => {
-        // By user_name
-        const userNameCompare = first.user.user_name.localeCompare(
-            second.user.user_name
-        );
-        // By role_name
-        const roleNameCompare = first.role.role_name.localeCompare(
-            second.role.role_name
-        );
-        return userNameCompare != 0 ? userNameCompare : roleNameCompare;
-    });
-    return userRoleAssignments;
+    const sortedUserAccesses = userRoleAssignments
+        .slice()
```

Comment:
Hello, Ian! "slice()" is not needed indeed but "slice().sort()" is a good way of sorting without mutating the original array

Reply:
I think the only difference is if we want to change the original userRoleAssignments array. `slice()` will not change the original array, I feel in our case we don't need to keep the original array but also doesn't matter.

Reply:
Hi Jota,
I think, no "good" when there is no real purpose; it only is good when you need the original array.
This would add more confusion and reduce performance.